### DISCUSSION TITLE
Installazione Rust

### DISCUSSION THREAD

**Discussion by sp2003** [indomus/forum#897] [HUB: Home Assistant]
buondì

per poter aggiornare HA Core, è necessario installare il rust compiler, e qui iniziano i miei problemi.
Sia in ambiente venv che fuori, l'installazione con il `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`
non va a buon fine, in pratica si blocca all'installazione di rustc riportando
```
info: installing component 'rustc'
 91.6 MiB / 105.2 MiB ( 87 %)   2.6 MiB/s in 43s ETA:  5sKilled
```

la percentuale di installazione varia di volta in volta, sono anche arrivato al 99%.

Ho provato allora ad installare dai pacchetti stand alone disponibili sul sito, e l'installazione sembra andare a buon fine 
```
pi@domotic:~ $ rustc --version
rustc 1.55.0 (c8dfcfe04 2021-09-06)
```
ma comunque l'installazione di cryptography `/usr/local/bin/python3.8 -m pip install cryptography==3.4.8`
non si conclude mai. In venv l'installazione finisce con errori e chiede l'installazione di rust, fuori da venv continua fino a bloccare il rasp per avere saturato la ram.

cosa potrei fare?

HW è un RPi3b+ e S.O. Raspibian Buster lite

**Comment by Scialla** [comment]
Come ti ho detto già dall'altra parte potrebbe dipendere dalla quantità di RAM come indicato nella ISSUE.
In ogni caso devi farlo dentro il venv non fuori

Prova ad aumentare la dimensione del file di swap e riprova ad installare rust con curl
